editor/src/clj/editor/gui_clipping.clj

```diff
@@ -589,51 +590,26 @@
   [(:node-id scene) (get-in scene [:renderable :user-data :clipping-state])])
 
 (defn- trickle-down-default-layers
-  "Perform inheritance of parent node layer to child nodes with no
-  layer assigned, and strip layer from clipper nodes."
+  "Perform inheritance of parent node layer to child nodes without layer."
   ([scene]
    (trickle-down-default-layers nil scene))
   ([default-layer scene]
    (let [layer (get-in scene [:renderable :layer-index])]
      (if (and layer (>= layer 0))
-       (cond-> (update scene :children #(mapv (partial trickle-down-default-layers layer) %))
-         (clipper-scene? scene)
-         (assoc-in [:renderable :layer-index] nil)) ; clipper nodes render in the null layer (first)
-       (cond-> (update scene :children #(mapv (partial trickle-down-default-layers default-layer) %))
-         (not (clipper-scene? scene))
-         (assoc-in [:renderable :layer-index] default-layer))))))
-
-(defn- scope-scenes
-  "Collect scenes in depth first pre order (parent visited before children), stopping at clipper nodes."
-  [scene]
-  (loop [children-stack (list (:children scene))
-         scenes []]
-    (cond
-      (not (seq children-stack))
-      scenes
-
-      (not (seq (first children-stack)))
-      (recur (rest children-stack) scenes)
-
-      true
-      (let [child (ffirst children-stack)
-            next-children (if (clipper-scene? child) [] (:children child))]
-        (recur (cons next-children (cons (rest (first children-stack)) (rest children-stack)))
-               (conj scenes child))))))
```

Comment:
After the change, clipper nodes are treated no differently than regular gui nodes, so this entire function can be replaced with a `tree-seq` call below.